### DISCUSSION TITLE
Search on eloquent with nullable relation.

### DISCUSSION THREAD

**Discussion by tcagkansokmen** [yajra/laravel-datatables#2548] [Q&A]
I'm using server-side rendering, and trying to search between columns.

I got relation like below,

Post::with(['address']);
Post can have an address, but also It may not have any address too.

So that, my logic fails on there;

I'm using column like

{ data: 'address.name', name:'address.name'  }

so that, if there's no data inside "address relation" datatable return error, because it says "requested unknown parameter 'address.name'" because name field is undefined.

What could be the solve?

**Comment by RayhanYulanda** [comment]
+1 Have same issue

**Comment by yajra** [comment]
Since it's nullable, you should add a `withDefault()` on your relation.

```php
class Post extends Model
{
  public function address() {
    return $this->belongsTo(Address::class)->withDefault();
  }
}
```

**Reply by RayhanYulanda** [reply]
I've tried but still can't work. Still got an error `requested unknown parameter`

**Reply by RayhanYulanda** [reply]
I solved it add the "name" to the with default like this reference https://github.com/yajra/laravel-datatables/issues/2366#issuecomment-616985844

**Comment by dyaskur** [comment]
You can set default using datatable too using `columnDefs`

example:
```
     $('#page-length-option').DataTable({
                processing: true,
                serverSide: true,
                responsive: true,
                "columnDefs": [{
                    "targets": '_all',
                    "defaultContent": "self"
                }],
                ajax: "{{ url('activity/'.$user->id.'/json') }}",
                columns: [
                    {data: 'id', name: 'id', visible: false},
                    {data: 'description', name: 'description'},
                    {data: 'properties', name: 'properties'},
                    {data: 'subject.name', name: 'subject.name'},
                    {data: 'created_at', name: 'created_at'},
                ]
            });
```
